go/fn/examples/example_asmain_runner_test.go

```diff
@@ -19,43 +19,108 @@ import (
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 )
 
+var _ fn.Runner = &CustomFnConfig{}
+
+type CustomFnConfig struct {
+	Owner string
+	Org   string
+}
+
+// Run is the main function logic.
+// `ctx` provides easy methods to add info, error or warning result to `ResourceList.Results`.
+// `items` is parsed from the STDIN "ResourceList.Items".
+// `functionConfig` is from the STDIN "ResourceList.FunctionConfig". The value is parsed from a `CustomFnConfig` type
+// "owner" and "org" field.
+func (r *CustomFnConfig) Run(ctx *fn.Context, functionConfig *fn.KubeObject, items fn.KubeObjects) {
```

Comment:
Do we need `functionConfig`?

Reply:
I'm thinking about some special handling. Like set-namespace supports variant constructor by accepting a `kptfile.kpt.dev` named "ConfigMap"